packages/core/src/actions.ts

```diff
@@ -713,7 +718,7 @@ export function resolveActions<TContext, TEvent extends EventObject>(
             if (exec && preservedContexts) {
               const contextIndex = preservedContexts.length - 1;
               resolvedActionObject.exec = (_ctx, ...args) => {
-                exec?.(preservedContexts[contextIndex], ...args);
+                return exec?.(preservedContexts[contextIndex], ...args);
```

Comment:
Important to return here because the `asEffect*` implementation in `@xstate/react` needs to know the return value

Reply:
This will create yet another mismatch between `xstate` & `@xstate/react` versions (peer dependencies 😭  )

Reply:
Are you saying this shouldn't go in? Because this fixes `@xstate/react`. Previous versions of `xstate` worked, and this change "restores" the behavior to those previous versions.

Reply:
the problem that I'm seeing is that, according to your words, `@xstate/react` needs to know the return value - which in turn means that it starts to require a future release of XState, while it states that it works with `^4.11.0`:
https://github.com/statelyai/xstate/blob/10944fa6195a3146c7c120d7c3c5c5defeff4e72/packages/xstate-react/package.json#L50

It's a technically OK thing to do to **only** update `@xstate/react` without updating `xstate` - because that's a contract that we have defined, by using peer dependencies, version ranges, and so on. But, from what I understand, this is not true anymore here - one can have a functioning pair of `xstate` and `@xstate/react` and they might break their app by upgrading only `@xstate/react` (which should be OK to do).

So the question would be - is there a way not to break that contract here?